src/utils/useOverlayWidth.js

```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from "react";
+import { useSelector } from "react-redux";
+
+const useOverlayWidth = () => {
+  const overlayElement = useSelector((state) => state.app.overlayElement);
+  const [width, setWidth] = useState(0);
+  useEffect(() => {
+    const resizeObserver = new ResizeObserver((entries) => {
+      setWidth(entries[0].contentRect.width);
+    });
+    if (!overlayElement) {
+      resizeObserver.disconnect();
+      setWidth(0);
+      return;
+    }
+    resizeObserver.observe(overlayElement);
+  }, [overlayElement]);
```

Comment:
always return a function when you unlisten the resize

Reply:
Done